### PR TITLE
[DOCS] Reformats search template and multi search template APIs

### PR DESCRIPTION
**Pull request by szabosteve**
Relates to elastic/docs#937 and https://github.com/elastic/elasticsearch/issues/45621.

This PR updates the search template and multi search template APIs to align with the new [API reference template](https://github.com/elastic/docs/blob/master/shared/api-ref-ex.asciidoc).

Resources:
* [Search template API doc](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-template.html)
* [Multi search template API doc](https://www.elastic.co/guide/en/elasticsearch/reference/current/multi-search-template.html)
* [Search template API spec](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/search_template.json)

### CODE REVIEW

**Comment by szabosteve** [comment]
@elasticmachine update branch

File: docs/reference/search/search-template.asciidoc [hunk 1]
```diff
@@ -1,8 +1,7 @@
 [[search-template]]
 === Search Template
 
-The `/_search/template` endpoint allows to use the mustache language to pre render search requests,
-before they are executed and fill existing templates with template parameters.
+Allows to use the mustache language to pre render search requests.
```
Comment (jrodewig) [inline docs/reference/search/search-template.asciidoc]:
```suggestion
Allows you to use the mustache language to pre render search requests.
```

File: docs/reference/search/search-template.asciidoc [hunk 2]
```diff
@@ -22,19 +21,87 @@ GET _search/template
...
+[[search-template-api-request]]
+==== {api-request-title}
+
+`GET _search/template`
+
+
+[[search-template-api-desc]]
+==== {api-description-title}
+
+The `/_search/template` endpoint allows to use the mustache language to pre
```
Comment (jrodewig) [inline docs/reference/search/search-template.asciidoc]:
```suggestion
The `/_search/template` endpoint allows you to use the mustache language to pre- 
```

File: docs/reference/search/search-template.asciidoc [hunk 3]
```diff
@@ -22,19 +21,87 @@ GET _search/template
...
+`ignore_throttled`::
+  (Optional, boolean) If `true`, specified concrete, expanded or aliased indices 
+  are not included in the response when throttled. Defaults to `false`. 
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=index-ignore-unavailable]
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=preference]
+
+`profile`::
+  (Optional, boolean) If `true`, the query execution will be profiled. Defaults
```
Comment (jrodewig) [inline docs/reference/search/search-template.asciidoc]:
```suggestion
  (Optional, boolean) If `true`, the query execution is profiled. Defaults 
```

File: docs/reference/search/search-template.asciidoc [hunk 4]
```diff
@@ -22,19 +21,87 @@ GET _search/template
...
+include::{docdir}/rest-api/common-parms.asciidoc[tag=index]
+
+
+[[search-template-api-query-params]]
+==== {api-query-parms-title}
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=allow-no-indices]
+
+`ccs_minimize_roundtrips`::
+  (Optional, boolean) If `true`, network round-trips will be minimized as part
```
Comment (jrodewig) [inline docs/reference/search/search-template.asciidoc]:
```suggestion
  (Optional, boolean) If `true`, network round-trips are minimized for 
```

File: docs/reference/search/search-template.asciidoc [hunk 5]
```diff
@@ -22,19 +21,87 @@ GET _search/template
...
+
+
+[[search-template-api-query-params]]
+==== {api-query-parms-title}
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=allow-no-indices]
+
+`ccs_minimize_roundtrips`::
+  (Optional, boolean) If `true`, network round-trips will be minimized as part 
+  of cross-cluster search requests execution. Defaults to `true`.
```
Comment (jrodewig) [inline docs/reference/search/search-template.asciidoc]:
```suggestion
 cross-cluster search requests. Defaults to `true`.
```

File: docs/reference/search/search-template.asciidoc [hunk 6]
```diff
@@ -22,19 +21,87 @@ GET _search/template
...
+include::{docdir}/rest-api/common-parms.asciidoc[tag=index-ignore-unavailable]
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=preference]
+
+`profile`::
+  (Optional, boolean) If `true`, the query execution will be profiled. Defaults 
+  to `false`.
+
+`rest_total_hits_as_int`::
+  (Optional, boolean) If `true`, `hits.total` will be rendered as an integer in
```
Comment (jrodewig) [inline docs/reference/search/search-template.asciidoc]:
```suggestion
  (Optional, boolean) If `true`, `hits.total` aree rendered as an integer in 
```

File: docs/reference/search/search-template.asciidoc [hunk 7]
```diff
@@ -22,19 +21,87 @@ GET _search/template
...
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=preference]
+
+`profile`::
+  (Optional, boolean) If `true`, the query execution will be profiled. Defaults 
+  to `false`.
+
+`rest_total_hits_as_int`::
+  (Optional, boolean) If `true`, `hits.total` will be rendered as an integer in 
+  the rest search response. Defaults to `false`.
```
Comment (jrodewig) [inline docs/reference/search/search-template.asciidoc]:
```suggestion
  the response. Defaults to `false`.
```

File: docs/reference/search/search-template.asciidoc [hunk 8]
```diff
@@ -22,19 +21,87 @@ GET _search/template
...
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=routing]
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=scroll]
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=search_type]
+
+`typed_keys`::
+  (Optional, boolean) If `true`, aggregation and suggester names will be
```
Comment (jrodewig) [inline docs/reference/search/search-template.asciidoc]:
```suggestion
  (Optional, boolean) If `true`, aggregation and suggester names are 
```

File: docs/reference/search/search-template.asciidoc [hunk 9]
```diff
@@ -22,19 +21,87 @@ GET _search/template
...
+
+`typed_keys`::
+  (Optional, boolean) If `true`, aggregation and suggester names will be 
+  prefixed by their respective types in the response. Defaults to `false`.
+  
+
+[[search-template-api-request-body]]
+==== {api-request-body-title}
+
+The API request body must contain the search definition template and its params.
```
Comment (jrodewig) [inline docs/reference/search/search-template.asciidoc]:
```suggestion
The API request body must contain the search definition template and its parameters.
```

**Review by jrodewig — APPROVED** [review]
LGTM. Few editorial suggestions but not blocking.

**Comment by szabosteve** [comment]
@elasticmachine update branch

File: docs/reference/search/search-template.asciidoc [hunk 10]
```diff
@@ -112,24 +181,11 @@ DELETE _scripts/<templateid>
 
 //////////////////////////
```
Comment (jrodewig) [inline docs/reference/search/search-template.asciidoc]: Delete this comment. This is causing the CI to fail.

Comment (szabosteve) [inline docs/reference/search/search-template.asciidoc]:
```suggestion

```

**Comment by szabosteve** [comment]
Backported to 7.x, 7.4, and 7.3.

**Pull request merged**